src/platform/plugins/shared/dashboard/server/content_management/v3/transforms/out/control_group.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { defaults, flow } from 'lodash';
+import {
+  ControlGroupChainingSystem,
+  ControlLabelPosition,
+  DEFAULT_AUTO_APPLY_SELECTIONS,
+  DEFAULT_CONTROL_CHAINING,
+  DEFAULT_CONTROL_LABEL_POSITION,
+  DEFAULT_IGNORE_PARENT_SETTINGS,
+  type ParentIgnoreSettings,
+} from '@kbn/controls-plugin/common';
+import type { DashboardSavedObjectAttributes } from '../../../../dashboard_saved_object';
+import type { ControlGroupAttributes } from '../../types';
+import { transformControlsState } from './control_state';
+
+export const transformControlGroupOut: (
+  controlGroupInput: NonNullable<DashboardSavedObjectAttributes['controlGroupInput']>
+) => ControlGroupAttributes = flow(
+  transformControlGroupSetDefaults,
+  transformControlGroupProperties
+);
+
+// TODO We may want to remove setting defaults in the future
+function transformControlGroupSetDefaults(
+  controlGroupInput: NonNullable<DashboardSavedObjectAttributes['controlGroupInput']>
+) {
+  return defaults(controlGroupInput, {
+    controlStyle: DEFAULT_CONTROL_LABEL_POSITION,
+    chainingSystem: DEFAULT_CONTROL_CHAINING,
+    showApplySelections: !DEFAULT_AUTO_APPLY_SELECTIONS,
+  });
+}
+
+function transformControlGroupProperties({
+  controlStyle,
+  chainingSystem,
+  showApplySelections,
+  ignoreParentSettingsJSON,
+  panelsJSON,
+}: Required<
+  NonNullable<DashboardSavedObjectAttributes['controlGroupInput']>
+>): ControlGroupAttributes {
+  return {
+    labelPosition: controlStyle as ControlLabelPosition,
+    chainingSystem: chainingSystem as ControlGroupChainingSystem,
+    autoApplySelections: !showApplySelections,
+    ignoreParentSettings: ignoreParentSettingsJSON
+      ? flow(
+          JSON.parse,
+          transformIgnoreParentSettingsSetDefaults,
+          transformIgnoreParentSettingsProperties
+        )(ignoreParentSettingsJSON)
+      : DEFAULT_IGNORE_PARENT_SETTINGS,
+    controls: panelsJSON ? transformControlsState(panelsJSON) : [],
+  };
+}
+
+// TODO We may want to remove setting defaults in the future
+function transformIgnoreParentSettingsSetDefaults(
+  ignoreParentSettings: ParentIgnoreSettings
+): ParentIgnoreSettings {
+  return defaults(ignoreParentSettings, DEFAULT_IGNORE_PARENT_SETTINGS);
+}
+
+function transformIgnoreParentSettingsProperties({
```

Comment:
Is this just a noop? Why do we need it?

Reply:
I'm using it to extract and provide only __supported__ properties from the saved object. It guards against any unsupported properties that may exist in the saved object. I may be overly cautious here. 🤷